sklearn/utils/_response.py

```diff
@@ -72,15 +72,17 @@ def _get_response_values(
     if is_classifier(estimator):
         prediction_method = _check_response_method(estimator, response_method)
         classes = estimator.classes_
-        target_type = "binary" if len(classes) <= 2 else "multiclass"
-
-        if pos_label is not None and pos_label not in classes.tolist():
-            raise ValueError(
-                f"pos_label={pos_label} is not a valid label: It should be "
-                f"one of {classes}"
-            )
-        elif pos_label is None and target_type == "binary":
-            pos_label = pos_label if pos_label is not None else classes[-1]
+        if isinstance(classes, list):
+            target_type = "multilabel-indicator"
```

Comment:
For performance reasons, @thomasjpfan did not want to call it the first time. It involves calling `np.unique` and since we already have `classes_` from the estimator, then we don't need to.

Reply:
Isn't `classes_` going to be super small? Even a problem with a lot of classes will have less than a 100 unique classes no? Performance might not be a problem here.

Reply:
The unique operation is on the full `y`.

Reply:
Although I do not like it, running `type_of_target(classes)` seems okay.

Reply:
indeed `type_of_target` only on the classes would avoid spending to much time on the unique.